builtin/providers/aws/resource_aws_cloudfront_web_distribution.go

```diff
@@ -0,0 +1,587 @@
+
+	originId := fmt.Sprintf("%s-origin", d.Get("origin_domain_name"))
+	aliases := resourceAwsCloudFrontWebDistributionAwsStringLists(
+		d.Get("aliases"))
+	geoRestrictions := resourceAwsCloudFrontWebDistributionAwsStringLists(
+		d.Get("geo_restrictions"))
+	allowedMethods := resourceAwsCloudFrontWebDistributionHandleMethods(
+		d.Get("allowed_methods"))
+	cachedMethods := resourceAwsCloudFrontWebDistributionHandleMethods(
+		d.Get("cached_methods"))
+	forwardedHeaders := resourceAwsCloudFrontWebDistributionAwsStringLists(
+		d.Get("forwarded_headers"))
+	cookies := resourceAwsCloudFrontWebDistributionCookies(
+		d.Get("forward_cookie"), d.Get("whitelisted_cookies"))
+	viewerCertificate := &cloudfront.ViewerCertificate{
+		MinimumProtocolVersion: aws.String(d.Get("minimum_ssl").(string)),
+		SSLSupportMethod:       aws.String(d.Get("ssl_support_method").(string)),
+	}
+	if d.Get("certificate_id") == "" {
+		viewerCertificate.CloudFrontDefaultCertificate = aws.Bool(true)
+	} else {
+		viewerCertificate.IAMCertificateId = aws.String(d.Get("certificate_id").(string))
+	}
+
+	// PUT DistributionConfig requires, unlike POST, EVERY possible option to be set.
+	// Except for the configurable options, these are the defaults options.
+	return &cloudfront.DistributionConfig{
+		CallerReference:   callerReference,
+		Enabled:           aws.Bool(d.Get("enabled").(bool)),
+		Comment:           aws.String(d.Get("comment").(string)),
+		PriceClass:        aws.String(d.Get("price_class").(string)),
+		DefaultRootObject: aws.String(d.Get("default_root_object").(string)),
+		Aliases: &cloudfront.Aliases{
+			Quantity: aws.Int64(int64(len(aliases))),
+			Items:    aliases,
+		},
+		Origins: &cloudfront.Origins{
+			Quantity: aws.Int64(1),
+			Items: []*cloudfront.Origin{
+				&cloudfront.Origin{
+					DomainName: aws.String(d.Get("origin_domain_name").(string)),
+					Id:         aws.String(originId),
+					OriginPath: aws.String(d.Get("origin_path").(string)),
+					CustomOriginConfig: &cloudfront.CustomOriginConfig{
+						HTTPPort:             aws.Int64(int64(d.Get("origin_http_port").(int))),
+						HTTPSPort:            aws.Int64(int64(d.Get("origin_https_port").(int))),
+						OriginProtocolPolicy: aws.String(d.Get("origin_protocol_policy").(string)),
+					},
+				},
+			},
+		},
+		ViewerCertificate: viewerCertificate,
+		Logging: &cloudfront.LoggingConfig{
+			Enabled:        aws.Bool(d.Get("logging_enabled").(bool)),
+			IncludeCookies: aws.Bool(d.Get("logging_include_cookies").(bool)),
+			Prefix:         aws.String(d.Get("logging_prefix").(string)),
+			Bucket:         aws.String(d.Get("logging_bucket").(string)),
+		},
+		Restrictions: &cloudfront.Restrictions{
+			GeoRestriction: &cloudfront.GeoRestriction{
+				Quantity:        aws.Int64(int64(len(geoRestrictions))),
+				RestrictionType: aws.String(d.Get("geo_restriction_type").(string)),
+				Items:           geoRestrictions,
+			},
+		},
+		DefaultCacheBehavior: &cloudfront.DefaultCacheBehavior{
```

Comment:
Unless there's a reason why to keep it that way, can you move all of the fields that are related to `DefaultCacheBehavior` here into a sub-resource in the schema - similarly to `logging_*`?